geoengine/workflow.py

```diff
@@ -438,3 +430,64 @@ def workflow_by_id(workflow_id: UUID) -> Workflow:
     # TODO: check that workflow exists
 
     return Workflow(WorkflowId(workflow_id))
+
+
+class Colorizer:
+    '''
+    Creates a colormap and returns its JSON string
+    '''
+
+    __colormap_name: str
+    __colorizer_min_max: Tuple[float, float]
+    __steps: int
+    __no_data_color: Tuple[float, float, float, float]
+
+    def __init__(self,
+                 colorizer_min_max: Tuple[float, float],
+                 colormap_name: str = "viridis",
+                 steps: int = 16,
+                 no_data_color: Tuple[float, float, float, float] = (0, 0, 0, 0)) -> None:
+        '''Initialize a new 'Colorizer' object'''
+        self.__colorizer_min_max = colorizer_min_max
+        self.__colormap_name = colormap_name
+        self.__steps = steps
+        self.__no_data_color = no_data_color
+
+    def __repr__(self) -> str:
+        return f'Colorizer({self.__colorizer_min_max}, {self.__colormap_name}, {self.__steps}, {self.__no_data_color})'
+
+    def __str__(self) -> str:
+        return f'Colorizer: min_max: {self.__colorizer_min_max}, colormap: {self.__colormap_name}, ' \
+            f'steps: {self.__steps}, no_data_color: {self.__no_data_color}'
+
+    def colorizer_get_url(self) -> str:
+        '''
+        Returns url for custom colorizer
+        '''
+        cmap = cm.get_cmap(self.__colormap_name, 256)
+        gradient = cmap(np.linspace(0, 1, self.__steps))
+        vals = np.linspace(self.__colorizer_min_max[0], self.__colorizer_min_max[1], self.__steps)
+
+        colors = []
+        for enum in enumerate(gradient):
+            color = {"value": vals[enum[0]], "color": np.round((gradient[enum[0]] * 255), 2).astype(np.uint8)}
+            colors.append(color)
+
+        colorizer = ''
+        colorizer = 'custom:' + json.dumps({
+            "type": "linearGradient",
```

Comment:
Could you change the `Colorizer` class to have a `Colorizer` base class and a `LinearGradient` subclass?
We have more colorizer variants in the backend, so this makes sense to already split now.